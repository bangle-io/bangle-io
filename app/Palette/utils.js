import React from 'react';

export function addBoldToTitle(string, query) {
  if (!query) {
    return string;
  }

  let newString = string.split(query);

  return newString.flatMap((r, i) => {
    if (i === 0) {
      return r;
    }
    return [<b key={i}>{query}</b>, r];
  });
}
