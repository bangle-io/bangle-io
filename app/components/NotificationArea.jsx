import React, { useContext } from 'react';
import {
  ButtonIcon,
  CloseIcon,
  ExclamationIcon,
  InformationCircleIcon,
  ExclamationCircleIcon,
  CheckCircleIcon,
} from 'ui-components/index';
import { UIManagerContext } from 'ui-context';

export function NotificationArea({}) {
  const { notifications, dispatch } = useContext(UIManagerContext);
  return (
    <div className="z-50 fixed bottom-0 right-0">
      {notifications.map((n) => (
        <Notification
          key={n.uid}
          uid={n.uid}
          onDismiss={() => {
            dispatch({
              type: 'UI/DISMISS_NOTIFICATION',
              value: {
                uid: n.uid,
              },
            });
          }}
          content={n.content}
          buttons={n.buttons}
          severity={n.severity}
        />
      ))}
    </div>
  );
}

const Severity = {
  error: () => (
    <ExclamationCircleIcon style={{ color: 'var(--severity-error-color)' }} />
  ),
  warning: () => (
    <ExclamationIcon style={{ color: 'var(--severity-warning-color)' }} />
  ),
  info: () => (
    <InformationCircleIcon style={{ color: 'var(--severity-info-color)' }} />
  ),
  success: () => (
    <CheckCircleIcon style={{ color: 'var(--severity-success-color)' }} />
  ),
};

export function Notification({ content = '', buttons, severity, onDismiss }) {
  return (
    <div
      className="relative shadow my-4 mx-4 w-96 p-2 duration-100 ease-in-out transition"
      style={{
        backgroundColor: 'var(--bg-stronger-color)',
        boxShadow: '0px 0px 4px 2px rgba(0, 0, 0, 0.15)',
      }}
    >
      <div className="w-full flex flex-row ">
        <span className="mr-2">{Severity[severity]()}</span>
        <div className="w-full text-sm">
          {typeof content === 'string' ? <span>{content}</span> : content}
        </div>
        <div>
          <ButtonIcon
            hint="dismiss"
            hintPos="left"
            onClick={async (e) => {
              onDismiss();
            }}
          >
            <CloseIcon style={{ height: 16, width: 16 }} />
          </ButtonIcon>
        </div>
      </div>
      <div className="w-full flex flex-row-reverse mt-3">
        {buttons &&
          buttons.map((b, i) => <React.Fragment key={i}>{b}</React.Fragment>)}
      </div>
    </div>
  );
}
