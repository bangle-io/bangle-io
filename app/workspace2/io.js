import * as idb from 'idb-keyval';

export class FileOps {
  // file
  static async getFile(wsPath) {
    return idb.get(wsPath);
  }

  static async deleteFile(wsPath) {
    const file = await this.getFile(wsPath);
    if (!file) {
      throw new FileError(`File ${wsPath} does not exists`);
    }
    return idb.del(wsPath);
  }

  static async renameFile(wsPath, newWsPath) {
    const file = await this.getFile(wsPath);

    if (!file) {
      throw new FileError(`File ${wsPath} does not exists`);
    }

    const newFile = await this.getFile(newWsPath);

    if (newFile) {
      throw new FileError(`File ${newWsPath} already exists`);
    }

    await this.deleteFile(wsPath);
    return this.createFile(newWsPath, file);
  }

  static async createFile(wsPath, payload) {
    if (await this.getFile(wsPath)) {
      throw new FileError(`File ${wsPath} already exists`);
    }
    return idb.set(wsPath, payload);
  }

  static async updateFile(wsPath, payload) {
    if (!(await this.getFile(wsPath))) {
      throw new FileError(`File ${wsPath} does not exists`);
    }
    return idb.set(wsPath, payload);
  }

  static async listFiles(wsName) {
    const keys = (await idb.keys()) || [];

    return keys.filter((k) => k.startsWith(wsName + ':'));
  }
}

export class FileError extends Error {
  constructor(message) {
    super('FileError: ' + message);
    this.name = 'FileError';
  }
}
export class FilePermissionError extends Error {
  constructor(message) {
    super('FilePermissionError: ' + message);
    this.name = 'FilePermissionError';
  }
}
