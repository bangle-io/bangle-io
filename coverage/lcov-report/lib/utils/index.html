
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for lib/utils</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> lib/utils</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">31.72% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>46/145</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">21.53% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>14/65</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">14.7% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>5/34</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">31.25% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>45/144</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="cx.ts"><a href="cx.ts.html">cx.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

<tr>
	<td class="file high" data-value="index.ts"><a href="index.ts.html">index.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file low" data-value="safe-js.ts"><a href="safe-js.ts.html">safe-js.ts</a></td>
	<td data-value="25.88" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 25%"></div><div class="cover-empty" style="width: 75%"></div></div>
	</td>
	<td data-value="25.88" class="pct low">25.88%</td>
	<td data-value="85" class="abs low">22/85</td>
	<td data-value="28.26" class="pct low">28.26%</td>
	<td data-value="46" class="abs low">13/46</td>
	<td data-value="7.69" class="pct low">7.69%</td>
	<td data-value="26" class="abs low">2/26</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="84" class="abs low">21/84</td>
	</tr>

<tr>
	<td class="file low" data-value="window.ts"><a href="window.ts.html">window.ts</a></td>
	<td data-value="28" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 28%"></div><div class="cover-empty" style="width: 72%"></div></div>
	</td>
	<td data-value="28" class="pct low">28%</td>
	<td data-value="50" class="abs low">14/50</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="18" class="abs low">0/18</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="7" class="abs low">2/7</td>
	<td data-value="28" class="pct low">28%</td>
	<td data-value="50" class="abs low">14/50</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2023-12-02T21:35:16.931Z
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    