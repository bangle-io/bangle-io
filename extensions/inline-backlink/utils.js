export function removeMdExtension(str) {
  if (str.endsWith('.md')) {
    return str.slice(0, -3);
  }
  return str;
}
