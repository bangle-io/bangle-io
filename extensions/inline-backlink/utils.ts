import {
  filePathToWsPath,
  removeExtension,
  resolvePath,
  sanitizeFilePath,
  suffixWithNoteExtension,
  validateWsPath,
} from '@bangle.io/ws-path';

export function wsPathFromQuery(query: string, wsName: string) {
  let filePath = query.split(']]').join('');
  filePath = filePath.trim();
  filePath = suffixWithNoteExtension(filePath);
  let wsPath = filePathToWsPath(wsName, filePath);
  try {
    validateWsPath(wsPath);
  } catch (error) {
    wsPath = filePathToWsPath(wsName, sanitizeFilePath(filePath));
  }

  return wsPath;
}

export function getBacklinkPath(wsPath: string, allWsPaths: readonly string[]) {
  const { fileName, filePath } = resolvePath(wsPath);
  const matchingFilenames = allWsPaths.filter(
    (w) => resolvePath(w).fileName === fileName,
  );

  if (matchingFilenames.length === 0) {
    return removeExtension(filePath);
  }

  // if there are multiple files with the same name
  // give it an absolute path
  if (matchingFilenames.length > 1) {
    return removeExtension(filePath);
  }

  return removeExtension(fileName);
}
