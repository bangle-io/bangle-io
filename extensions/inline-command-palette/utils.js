export function replaceSuggestText(newText) {
  return (state, dispatch, view) => {};
}
