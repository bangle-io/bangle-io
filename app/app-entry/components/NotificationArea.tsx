import React from 'react';

import { useSerialOperationContext } from '@bangle.io/api';
import { useNsmSlice } from '@bangle.io/bangle-store-context';
import { SEVERITY } from '@bangle.io/constants';
import type { NotificationPayloadType } from '@bangle.io/shared-types';
import { nsmNotification } from '@bangle.io/slice-notification';
import {
  Button,
  BUTTON_VARIANT,
  CheckCircleIcon,
  CloseIcon,
  ExclamationCircleIcon,
  ExclamationIcon,
  InformationCircleIcon,
} from '@bangle.io/ui-components';
import { useInterval } from '@bangle.io/utils';

const CLEAR_INTERVAL = 8000;

export function NotificationArea() {
  const [{ notifications }, dispatch] = useNsmSlice(
    nsmNotification.nsmNotificationSlice,
  );

  useInterval(
    () => {
      let currentTime = Date.now();
      let toRemove = notifications
        .filter((n) => {
          return (
            n.transient &&
            n.createdAt &&
            CLEAR_INTERVAL < currentTime - n.createdAt
          );
        })
        .map((n) => {
          return n.uid;
        });

      if (toRemove.length > 0) {
        dispatch(nsmNotification.dismissNotification(toRemove));
      }
    },
    [notifications, dispatch],
    2000,
  );

  return (
    <div className="fixed bottom-0 right-0 z-50">
      {notifications.map((n) => (
        <Notification
          key={n.uid}
          onDismiss={() => {
            dispatch(nsmNotification.dismissNotification(n.uid));
          }}
          title={n.title}
          content={n.content?.split('\n').map((r, i) => (
            <span key={i}>{r}</span>
          ))}
          severity={n.severity}
          buttons={n.buttons}
        />
      ))}
    </div>
  );
}

const SeverityMap: Record<
  Exclude<NotificationPayloadType['severity'], undefined>,
  () => React.ReactNode
> = {
  error: () => (
    <ExclamationCircleIcon className="w-6 h-6 color-colorCriticalIcon" />
  ),
  warning: () => <ExclamationIcon className="w-6 h-6 color-colorCautionIcon" />,
  info: () => (
    <InformationCircleIcon className="w-6 h-6 color-colorNeutralIcon" />
  ),
  success: () => (
    <CheckCircleIcon className="w-6 h-6 color-colorPositiveIcon" />
  ),
};

export function Notification({
  content = '',
  title,
  buttons,
  severity = SEVERITY.INFO,
  onDismiss,
}: {
  title?: string;
  content: React.ReactNode;
  severity?: NotificationPayloadType['severity'];
  buttons?: NotificationPayloadType['buttons'];
  onDismiss: () => void;
}) {
  const { dispatchSerialOperation } = useSerialOperationContext();

  return (
    <div
      data-testid="app-entry_notification"
      className="bg-colorBgLayerFloat w-96 relative p-2 mx-4 my-4 transition duration-100 ease-in-out shadow-xl rounded shadow-lg border-neutral"
    >
      <div className="flex flex-col w-full">
        <div className="flex flex-row">
          <div className="mr-2">{SeverityMap[severity]()}</div>
          <div className="flex-grow">{title}</div>
          <div>
            <Button
              size="xs"
              variant={BUTTON_VARIANT.TRANSPARENT}
              ariaLabel="dismiss notification"
              leftIcon={<CloseIcon />}
              onPress={() => {
                onDismiss();
              }}
            />
          </div>
        </div>
        <div className="w-full text-sm flex flex-col">
          {typeof content === 'string' ? <span>{content}</span> : content}
        </div>
      </div>
      <div className="flex flex-row-reverse w-full mt-3">
        {buttons &&
          buttons.map((b, i) => (
            <Button
              key={i}
              tooltipPlacement="left"
              className="ml-3"
              size="sm"
              tone={
                severity === 'error'
                  ? 'critical'
                  : severity === 'warning'
                  ? 'caution'
                  : severity === 'success'
                  ? 'positive'
                  : 'secondary'
              }
              variant={severity === 'info' ? 'soft' : 'solid'}
              onPress={async () => {
                if (b.dismissOnClick) {
                  onDismiss();
                }
                dispatchSerialOperation({ name: b.operation });
              }}
              ariaLabel={b.hint}
              text={b.title}
            />
          ))}
      </div>
    </div>
  );
}
