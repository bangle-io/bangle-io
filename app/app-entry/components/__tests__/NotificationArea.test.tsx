/**
 * @jest-environment @bangle.io/jsdom-env
 */
import { act, fireEvent, render } from '@testing-library/react';
import React from 'react';

import { useSliceState } from '@bangle.io/bangle-store-context';
import type { NotificationPayloadType } from '@bangle.io/shared-types';
import { sleep } from '@bangle.io/utils';

import { NotificationArea } from '../NotificationArea';

jest.mock('@bangle.io/bangle-store-context', () => {
  return { useSliceState: jest.fn() };
});

const useSliceStateMock = jest.mocked(useSliceState);

beforeEach(() => {
  useSliceStateMock.mockImplementation(() => ({
    store: { dispatch: jest.fn(), state: {} } as any,
    sliceState: { notifications: [] },
    dispatch: jest.fn(),
  }));
});

describe('NotificationArea', () => {
  test('renders empty', async () => {
    const result = await render(<NotificationArea />);

    expect(result.container).toMatchInlineSnapshot(`
      <div>
        <div
          class="fixed bottom-0 right-0 z-50"
        />
      </div>
    `);
  });

  test('renders with content', async () => {
    const uiDispatchMock = jest.fn();
    let notificationsObj: { notifications: NotificationPayloadType[] } = {
      notifications: [],
    };
    useSliceStateMock.mockImplementation(() => ({
      store: { dispatch: uiDispatchMock, state: {} } as any,
      sliceState: { notifications: notificationsObj.notifications },
      dispatch: jest.fn(),
    }));

    const result = await render(<NotificationArea />);

    expect(result.container).toMatchInlineSnapshot(`
      <div>
        <div
          class="fixed bottom-0 right-0 z-50"
        />
      </div>
    `);

    notificationsObj.notifications = [
      { uid: 'one', title: 'Title', content: 'hello you!' },
    ];

    expect(useSliceState).toBeCalledTimes(1);

    await result.rerender(<NotificationArea />);

    expect(useSliceState).toBeCalledTimes(2);
    expect(result.container.innerHTML.includes('hello you')).toBe(true);

    notificationsObj.notifications = [
      { uid: 'one', content: 'hello you!', title: 'Title One' },
      {
        title: 'Title Two',
        uid: 'second',
        content: 'second!',
        buttons: [
          {
            operation: 'operation::@bangle.io/test:something',
            title: 'Test Op',
            dismissOnClick: true,
            hint: 'test-btn',
          },
        ],
      },
    ];

    await result.rerender(<NotificationArea />);
    expect(useSliceState).toBeCalledTimes(3);

    expect(result.container.innerHTML.includes('hello you')).toBe(true);
    expect(result.container.innerHTML.includes('second!')).toBe(true);

    expect(result.container).toMatchSnapshot();

    expect(uiDispatchMock).toBeCalledTimes(0);

    fireEvent.click(result.getByLabelText('test-btn'));

    // should dismiss the notification since `dismissOnClick` is true
    expect(uiDispatchMock).toBeCalledTimes(1);
    expect(uiDispatchMock).toHaveBeenCalledWith({
      name: 'action::@bangle.io/slice-notification:DISMISS_NOTIFICATION',
      value: { uids: ['second'] },
    });
  });

  test('removes notification on clicking', async () => {
    const uiDispatchMock = jest.fn();
    let notificationsObj: { notifications: NotificationPayloadType[] } = {
      notifications: [],
    };
    useSliceStateMock.mockImplementation(() => ({
      store: { dispatch: uiDispatchMock, state: {} } as any,
      dispatch: jest.fn(),
      sliceState: { notifications: notificationsObj.notifications },
    }));
    const result = await render(<NotificationArea />);

    notificationsObj.notifications = [
      { uid: 'one', title: 'title', content: 'hello you!' },
    ];

    await result.rerender(<NotificationArea />);
    expect(useSliceState).toBeCalledTimes(2);

    expect(result.container.innerHTML.includes('hello you')).toBe(true);

    const prom = sleep();
    fireEvent.click(result.getByLabelText('dismiss notification'));
    await act(() => prom);

    expect(uiDispatchMock).toBeCalledTimes(1);
    expect(uiDispatchMock).nthCalledWith(1, {
      name: 'action::@bangle.io/slice-notification:DISMISS_NOTIFICATION',
      value: {
        uids: ['one'],
      },
    });
  });
});
