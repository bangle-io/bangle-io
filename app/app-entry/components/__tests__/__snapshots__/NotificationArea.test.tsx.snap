// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationArea renders with content 2`] = `
<div>
  <div
    class="fixed bottom-0 right-0 z-50"
  >
    <div
      class="bg-colorBgLayerFloat w-96 relative p-2 mx-4 my-4 transition duration-100 ease-in-out shadow-xl rounded shadow-lg border-neutral"
      data-testid="app-entry_notification"
    >
      <div
        class="flex flex-col w-full"
      >
        <div
          class="flex flex-row"
        >
          <div
            class="mr-2"
          >
            <svg
              class="w-6 h-6 color-colorNeutralIcon"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
          <div
            class="flex-grow"
          >
            Title One
          </div>
          <div>
            <button
              aria-label="dismiss notification"
              class="text-xs font-600 min-w-6 h-6 px-1_5  select-none inline-flex justify-center items-center rounded-md whitespace-nowrap overflow-hidden py-1 transition-all duration-100 cursor-pointer "
              style="background-color: transparent;"
              type="button"
            >
              <span
                class="flex flex-grow-1 overflow-hidden "
                style="justify-content: center;"
              >
                <span>
                  <svg
                    class=" w-4 h-4  "
                    fill="currentColor"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.2253 4.81108C5.83477 4.42056 5.20161 4.42056 4.81108 4.81108C4.42056 5.20161 4.42056 5.83477 4.81108 6.2253L10.5858 12L4.81114 17.7747C4.42062 18.1652 4.42062 18.7984 4.81114 19.1889C5.20167 19.5794 5.83483 19.5794 6.22535 19.1889L12 13.4142L17.7747 19.1889C18.1652 19.5794 18.7984 19.5794 19.1889 19.1889C19.5794 18.7984 19.5794 18.1652 19.1889 17.7747L13.4142 12L19.189 6.2253C19.5795 5.83477 19.5795 5.20161 19.189 4.81108C18.7985 4.42056 18.1653 4.42056 17.7748 4.81108L12 10.5858L6.2253 4.81108Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </span>
            </button>
          </div>
        </div>
        <div
          class="w-full text-sm flex flex-col"
        >
          <span>
            hello you!
          </span>
        </div>
      </div>
      <div
        class="flex flex-row-reverse w-full mt-3"
      />
    </div>
    <div
      class="bg-colorBgLayerFloat w-96 relative p-2 mx-4 my-4 transition duration-100 ease-in-out shadow-xl rounded shadow-lg border-neutral"
      data-testid="app-entry_notification"
    >
      <div
        class="flex flex-col w-full"
      >
        <div
          class="flex flex-row"
        >
          <div
            class="mr-2"
          >
            <svg
              class="w-6 h-6 color-colorNeutralIcon"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
          <div
            class="flex-grow"
          >
            Title Two
          </div>
          <div>
            <button
              aria-label="dismiss notification"
              class="text-xs font-600 min-w-6 h-6 px-1_5  select-none inline-flex justify-center items-center rounded-md whitespace-nowrap overflow-hidden py-1 transition-all duration-100 cursor-pointer "
              style="background-color: transparent;"
              type="button"
            >
              <span
                class="flex flex-grow-1 overflow-hidden "
                style="justify-content: center;"
              >
                <span>
                  <svg
                    class=" w-4 h-4  "
                    fill="currentColor"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.2253 4.81108C5.83477 4.42056 5.20161 4.42056 4.81108 4.81108C4.42056 5.20161 4.42056 5.83477 4.81108 6.2253L10.5858 12L4.81114 17.7747C4.42062 18.1652 4.42062 18.7984 4.81114 19.1889C5.20167 19.5794 5.83483 19.5794 6.22535 19.1889L12 13.4142L17.7747 19.1889C18.1652 19.5794 18.7984 19.5794 19.1889 19.1889C19.5794 18.7984 19.5794 18.1652 19.1889 17.7747L13.4142 12L19.189 6.2253C19.5795 5.83477 19.5795 5.20161 19.189 4.81108C18.7985 4.42056 18.1653 4.42056 17.7748 4.81108L12 10.5858L6.2253 4.81108Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </span>
            </button>
          </div>
        </div>
        <div
          class="w-full text-sm flex flex-col"
        >
          <span>
            second!
          </span>
        </div>
      </div>
      <div
        class="flex flex-row-reverse w-full mt-3"
      >
        <button
          aria-label="test-btn"
          class="ml-3 text-sm font-600 h-8 min-w-8 px-2  select-none inline-flex justify-center items-center rounded-md whitespace-nowrap overflow-hidden py-1 transition-all duration-100 cursor-pointer "
          type="button"
        >
          <span
            class="flex flex-grow-1 overflow-hidden "
            style="justify-content: center;"
          >
            <span
              class="text-ellipsis overflow-hidden "
            >
              Test Op
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
